File: go/example_code/polly/pollyListLexicons.go
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package main

import (
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/polly"

	"fmt"
	"os"
)

func main() {
	// Initialize a session that the SDK uses to load
	// credentials from the shared credentials file. (~/.aws/credentials).
	sess := session.Must(session.NewSessionWithOptions(session.Options{
		SharedConfigState: session.SharedConfigEnable,
	}))

	// Create Polly client
	svc := polly.New(sess)

	resp, err := svc.ListLexicons(nil)
	if err != nil {
		fmt.Println("Got error calling ListLexicons:")
		fmt.Print(err.Error())
		os.Exit(1)
	}

	for _, l := range resp.Lexicons {
		fmt.Println(*l.Name)
		fmt.Println("  Alphabet: " + *l.Attributes.Alphabet)
		fmt.Println("  Language: " + *l.Attributes.LanguageCode)
		fmt.Println("")
	}
}

File: ruby/example_code/polly/polly_list_lexicons.rb
# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0

# Purpose:
# polly_list_lexicons.rb demonstrates how to
# list your Amazon Polly lexicons using the AWS SDK for Ruby.

# snippet-start:[polly.ruby.listLexicons]

require "aws-sdk-polly"  # In v2: require 'aws-sdk'

begin
  # Create an Amazon Polly client using
  # credentials from the shared credentials file ~/.aws/credentials
  # and the configuration (region) from the shared configuration file ~/.aws/config
  polly = Aws::Polly::Client.new

  resp = polly.list_lexicons

  resp.lexicons.each do |l|
    puts l.name
    puts "  Alphabet:" + l.attributes.alphabet
    puts "  Language:" + l.attributes.language
    puts
  end
rescue StandardError => ex
  puts "Could not get lexicons"
  puts "Error message:"
  puts ex.message
end
# snippet-end:[polly.ruby.listLexicons]
